### PR TITLE
locode-db: generate locode new command for locode.db generation

### LINKED ISSUE #6: v0.2.0 is broken: niko is not shown on geomap
`neofs_net_monitor_netmap{﻿location﻿=﻿"Vassilevsky Ostrov/St Petersburg"﻿}`

```
monitor: can't fetch 02a4da4d4dd51a682ab1a58caf0903361d3fd6b56622421d32bd140b37732c5cb8 geoposition, could not get record by provided locode
```

mona is downgraded to 0.1.0 now

### LINKED ISSUE #15: Move locode DB handling utility here
Move `pkg/util/locode` from neofs-node here, this code doesn't belong to NeoFS. Make sure we have some command here to generate the same DB.

### PR DESCRIPTION
**Pull request by AliceInHunterland**
Move `pkg/util/locode` from the neofs-node repository to this repository. A new command for locode.db is introduced here: locode-db generate.

Closes #15 .

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -0,0 +1,31 @@
+module github.com/nspcc-dev/locode-db
+
+go 1.20
+
+require (
+	github.com/nspcc-dev/neofs-node v0.38.1
+	github.com/nspcc-dev/neofs-sdk-go v1.0.0-rc.11
```
Comment (roman-khimov) [inline go.mod:7]: As well as this one.

File: cmd/exit.go [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+		awaitTimeout
+		alreadyRemoved
+	)
+
+	var code int
+	var accessErr = new(sdkstatus.ObjectAccessDenied)
+	var alreadyRemovedErr = new(sdkstatus.ObjectAlreadyRemoved)
+
+	switch {
+	case errors.Is(err, sdkstatus.ErrServerInternal):
```
Comment (roman-khimov) [inline cmd/exit.go:44]: We won't have these errors here, so this logic (and SDK dependency) doesn't make much sense.

File: cmd/locode/db/boltdb/calls.go [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+// Open opens an underlying BoltDB instance.
+//
+// Timeout of BoltDB opening is 3s (only for Linux or Darwin).
+//
+// Opens BoltDB in read-only mode if DB is read-only.
+func (db *DB) Open() error {
+	// copy-paste from metabase:
+	// consider universal Open/Close for BoltDB wrappers
+
+	err := util.MkdirAllX(filepath.Dir(db.path), db.mode)
```
Comment (roman-khimov) [inline cmd/locode/db/boltdb/calls.go:23]: Not needed even in the node, you shouldn't care about dirs (and shouldn't have neofs-node dependency).

File: cmd/locode/db/boltdb/calls.go [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+// Open opens an underlying BoltDB instance.
+//
+// Timeout of BoltDB opening is 3s (only for Linux or Darwin).
+//
+// Opens BoltDB in read-only mode if DB is read-only.
+func (db *DB) Open() error {
+	// copy-paste from metabase:
+	// consider universal Open/Close for BoltDB wrappers
+
+	err := os.MkdirAll(filepath.Dir(db.path), db.mode)
```
Comment (roman-khimov) [inline cmd/locode/db/boltdb/calls.go:23]: Ah, and please drop this logic completely. Just open a DB and expect dir to be present.

File: Makefile [hunk 5]
```diff
@@ -56,4 +60,5 @@ help:
 clean:
 	rm -f in/*
 	rm -f tmp/*
-	rm -f locode_db
+	rm -f $(LOCODEDB)
+	rm -rf bin
```
Comment (roman-khimov) [inline Makefile:64]: Missing newline

Comment (AliceInHunterland) [reply Makefile:64]: at the end of file?

File: Makefile [hunk 6]
```diff
@@ -56,4 +60,6 @@ help:
...
+
```
Comment (roman-khimov) [inline Makefile:65]: This one wasn't really needed, but OK.

**Review by roman-khimov — APPROVED** [review]

**Comment by roman-khimov** [comment]
There is some problem with DCO though.

**Review by roman-khimov — APPROVED** [review]

File: Makefile [hunk 7]
```diff
@@ -21,22 +22,25 @@ in/airports.dat:
 in/countries.dat:
 	wget -c https://raw.githubusercontent.com/jpatokal/openflights/master/data/countries.dat -O in/countries.dat
 
-in/continents.geojson: $(DIRS)
-	zcat continents.geojson.gz > in/continents.geojson
+geojson: continents.geojson.gz
```
Comment (carpawell) [inline Makefile:25]: what does `continents.geojson.gz` rule mean here?

File: Makefile [hunk 8]
```diff
@@ -21,22 +22,25 @@ in/airports.dat:
...
 unlocode:
-	wget -c https://service.unece.org/trade/locode/loc222csv.zip -O tmp/loc222csv.zip
-	unzip -u tmp/loc222csv.zip -d in/
+	wget -c https://service.unece.org/trade/locode/loc231csv.zip -O tmp/loc231csv.zip
+	unzip -u tmp/loc231csv.zip -d in/
 
-locode_db: unlocode in/continents.geojson in/airports.dat in/countries.dat
-	$(NEOFSCLI) util locode generate \
+bin/$(LOCODECLI):
+	go build -o $(LOCODECLI)
```
Comment (carpawell) [inline Makefile:33]: `.gitignore` file does not have `locode-db` file and it is built in the root, not in the `bin` or `tmp` directories. is it expected?

File: Makefile [hunk 9]
```diff
@@ -56,4 +60,6 @@ help:
 clean:
 	rm -f in/*
 	rm -f tmp/*
-	rm -f locode_db
+	rm -f $(LOCODEDB)
+	rm -rf bin
```
Comment (carpawell) [inline Makefile:64]: this dir is not created by us

File: Makefile [hunk 10]
```diff
@@ -56,4 +60,6 @@ help:
 clean:
 	rm -f in/*
 	rm -f tmp/*
-	rm -f locode_db
+	rm -f $(LOCODEDB)
```
Comment (carpawell) [inline Makefile:63]: it removes db but does not remove cli created when i call `make`

File: Makefile (hunk 7 above)
Comment (AliceInHunterland) [reply Makefile:25]: dependence on the existence of the file continents.geojson.gz

Comment (carpawell) [reply Makefile:25]: @roman-khimov, is it a good practice to place a dependency as a prerequisite but without any rule to create it? (i do not know)

Comment (roman-khimov) [reply Makefile:25]: It's OK since this file does exist in the repository. If it changes, the dependency will trigger a rebuild of the target.

**Pull request merged**